searx/engines/duckduckgo.py

```diff
@@ -63,66 +63,108 @@ def get_region_code(lang, lang_list=None):
 
 
 def request(query, params):
-    if params['time_range'] is not None and params['time_range'] not in time_range_dict:
-        return params
 
     params['url'] = url
     params['method'] = 'POST'
+
     params['data']['q'] = query
-    params['data']['b'] = ''
+
+    # The API is not documented, so we do some reverse engineering and emulate
+    # what https://lite.duckduckgo.com/lite/ does when you press "next Page"
+    # link again and again ..
+
+    params['headers']['Content-Type'] = 'application/x-www-form-urlencoded'
+
+    # initial page does not have an offset
+    if params['pageno'] == 2:
+        # second page does have an offset of 30
+        offset = (params['pageno'] - 1) * 30
+        params['data']['s'] = offset
+        params['data']['dc'] = offset + 1
+
+    elif params['pageno'] > 2:
+        # third and following pages do have an offset of 30 + n*50
+        offset = 30 + (params['pageno'] - 2) * 50
+        params['data']['s'] = offset
+        params['data']['dc'] = offset + 1
+
+    # initial page does not have additional data in the input form
+    if params['pageno'] > 1:
+        # request the second page (and more pages) needs 'o' and 'api' arguments
+        params['data']['o'] = 'json'
+        params['data']['api'] = 'd.js'
+
+    # initial page does not have additional data in the input form
+    if params['pageno'] > 2:
+        # request the third page (and more pages) some more arguments
+        params['data']['nextParams'] = ''
+        params['data']['v'] = ''
+        params['data']['vqd'] = ''
```

Comment:
Today I improved the paging support, I try to emulate what the input forms from https://lite.duckduckgo.com/lite/  do when you click on "next page" .. while looking along the implementation the "API" seems a bit strange, but this is what the input forms do ...